src/foam/dao/index/AltSelectPlan.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package foam.dao.index;
+
+import foam.core.FObject;
+
+import foam.dao.Sink;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Not;
+import foam.mlang.predicate.Predicate;
+
+public class AltSelectPlan implements SelectPlan {
+  protected Object state_;
+  protected Plan bestPlan_ = new NotFoundPlan();
```

Comment:
Make a SelectPlan to avoid casting below.

Reply:
Ok